cpp/log/strict_consistent_store.h

```diff
@@ -0,0 +1,93 @@
+#ifndef CERT_TRANS_LOG_STRICT_CONSISTENT_STORE_H_
+#define CERT_TRANS_LOG_STRICT_CONSISTENT_STORE_H_
+
+#include "log/consistent_store.h"
+#include "util/masterelection.h"
+
+namespace cert_trans {
+
+// A wrapper around a ConsistentStore which will not allow changes to
+// the cluster state which should only be performed by the current master
+// unless this node /is/ the current master.
+//
+// Note that while this is better than just gating the start of an action
+// (especially a long running action, e.g. a signing run) with a check to
```

Comment:
I'm re-reading this, and I think just adding "higher-level" to "gating the start of an action" should be sufficient to clarify this?